server/grpc/service.go

```diff
@@ -26,3 +27,16 @@ func (s *ServiceServer) Create(ctx context.Context, request *protobuf_api.Create
 	}
 	return &protobuf_api.CreateServiceResponse{Sid: srv.Sid, Hash: srv.Hash}, nil
 }
+
+// Delete deletes service by hash or sid.
+func (s *ServiceServer) Delete(ctx context.Context, request *protobuf_api.DeleteServiceRequest) (*protobuf_api.DeleteServiceResponse, error) {
+	// first, check if service has any running instances.
+	instances, err := s.sdk.Instance.GetAllByService(request.HashOrSid)
+	if err != nil {
+		return nil, err
+	}
+	if len(instances) > 0 {
+		return nil, errors.New("service has running instances. in order to delete the service, stop its instances first")
```

Comment:
@NicolasMahe you said before about listing running instance hashes here so devs can stop them. i think it's not relevant here, devs should do list running instances of service by using the cli